lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -526,6 +513,11 @@ static LogicalResult verifyFExtModuleOp(FExtModuleOp op) {
 
   if (!llvm::all_of(paramDict, checkParmValue))
     return failure();
+  auto argAttr = getFIRRTLModuleArgNameAttr(op);
+
+  if (op.getPorts().size() != argAttr.size()) {
+    return failure();
```

Comment:
Don't these need to emit errors before returning failure?

Reply:
This still needs to be something like `return op.emitError(...`, not `return failure()`.